KBV-192: Specify API with OpenAPI document

jkunle opened the pull request:
## Proposed changes

Specify an OpenAPI document for the defined API Gateway

### What changed

Experian KBV api is currently defined solely using SAM serverless API and Events, the change is to provide an OpenAPI document to describe this API

NB: Renamed resources in template file to distinguish them from KBV API

### Why did it change

<!-- Describe the reason these changes were made - the "why" -->

Defining an OpenAPI, would make the API easier to understand by consumers

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
Related to [KBV-101](https://govukverify.atlassian.net/browse/KBV-101) and  [KBV-102](https://govukverify.atlassian.net/browse/KBV-102) done
See also: [KBV-214](https://govukverify.atlassian.net/browse/KBV-101) yet to be done

This PR is based on [KBV-192](https://govukverify.atlassian.net/browse/KBV-192)

jkunle committed:
KBV-192 Describe by an OpenAPI document

idrop commented on deploy/api.yaml:
```diff
@@ -0,0 +1,188 @@
+  title: "Experian KBV Wrapper Api"
+  version: "1.0"
+x-amazon-apigateway-request-validators:
+  body:
```
can we wire in the `body` validator please? See https://github.com/alphagov/di-ipv-cri-fraud-api/blob/a2ca801bff1dc636ef8ad9df95f33bc20d31a28f/deploy/api.yaml#L33

jkunle committed:
KBV-192: Wire validator to endpoints

- Wired the aws request validator in the body of both endpoints
- Changed path name to use hyphens

idrop commented on lib/src/test/java/uk/gov/di/ipv/cri/experian/kbv/api/gateway/KBVGatewayTest.java:
```diff
@@ -54,7 +54,7 @@ void setUp() {
     @Test
     void shouldCallSubmitAnswersSuccessfully() throws InterruptedException {
-        // final String testRequestBody = "serialisedKbvApiRequest";
+        // final String testRequestBody = "serialisedExperianKbvApiRequest";
```
can we just remove this commented code instead of this commented change?

jkunle replied:
I'm planning to make another pass on some these test, there still more test coverage to be done generally, this a side-effect of find and replace

idrop commented on deploy/template.yaml:
```diff
@@ -12,9 +12,9 @@ Globals:
     VpcConfig:
       SecurityGroupIds:
-        - sg-03a1efaf2fed58442
+        - sg-0baf479c53ac54f59
```
ok to harcode this for now, but can we get this from an SSM param or a [mapping like in HKR ](https://github.com/alphagov/verify-hub-key-rotation/blob/6d2801dc8138e964617ac2980825450e4157a71c/template.yaml#L36)in future?

jkunle committed:
KBV-192: Parameterise lambda VPC config

idrop reviewed (approved):
Good for now @jkunle , thanks!
